module/icp/core/kcf_sched.c

```diff
@@ -1740,8 +1740,13 @@ kcf_last_req(void *last_req_arg, int status)
 		ct = (crypto_dual_data_t *)dcrops->dop_ciphertext;
 		break;
 	}
-	default:
-		break;
+	default: {
+		crypto_call_req_t *cr = &last_req->kr_callreq;
+
+		(*(cr->cr_callback_func))(cr->cr_callback_arg, status);
+		kmem_free(last_req, sizeof (kcf_dual_req_t));
+		return;
```

Comment:
It should probably be fatal. Most of the default -> breaks that were added were because Illumos didn't handlle all enum possibilities, but gcc on Linux wouldn't compile unless they were. This should probably assert or panic.